### PR TITLE
Upgrade dependencies and build abi3 wheels

### PR DESCRIPTION
**Pull request by messense**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -23,14 +23,14 @@ classifier = [
 provides-extra = ["validate"]
 
 [dependencies]
-pyo3 = { version = "0.12.3", features = ["extension-module"] }
-ncollide3d = "0.26.0"
+pyo3 = { version = "0.13.2", features = ["extension-module", "abi3-py36"] }
```
Comment (clbarnes) [inline Cargo.toml:26]: Haven't come across this before! Does it cost anything, or is it a hard "will compile/ won't compile" thing which happens to work for the subset of the abi ncollpyde uses?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -78,10 +77,11 @@ jobs:
...
       - uses: actions/setup-python@v2
         with:
-          python-version: ${{ matrix.python-version }}
-      - run: pip install $(grep ^maturin requirements.txt)
-      - if: startsWith(matrix.os,'ubuntu') && matrix.python-version == '3.8'
-        run: maturin publish -u __token__ -p ${{ secrets.MATURIN_PASSWORD }} --interpreter
-        name: Deploy sdist
-      - run: maturin publish -u __token__ -p ${{ secrets.MATURIN_PASSWORD }} -i python --no-sdist
+          python-version: '3.9'
+      - uses: messense/maturin-action@v1
```
Comment (clbarnes) [inline .github/workflows/ci.yaml:81]: With the abi3 feature, will this build a single wheel which can be used by all supported pythons, or will it still build different wheels and depend on either the GHA environment or the container environment for which pythons are available?

**Review by clbarnes — COMMENTED** [review]
Thanks for the contribution! I'd definitely prefer to use an action if possible and it looks like you've got sensible defaults for manylinux etc. (actually having some issues on that front with the current build process, see #9 ). The abi3 feature is a big plus too, as we're using such a small portion of the ABI as it is, so long as it can build.

File: Cargo.toml (hunk 1 above)
Comment (messense) [reply Cargo.toml:26]: It'll fail to compile if you use non-limited APIs.

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (messense) [reply .github/workflows/ci.yaml:81]: It'll only build a single wheel.

Comment (messense) [reply .github/workflows/ci.yaml:81]: And it's manylinux1 compliant.

```
📦 Built wheel for abi3 Python ≥ 3.6 to /home/runner/work/ncollpyde/ncollpyde/target/wheels/ncollpyde-0.15.2-cp36-abi3-manylinux_2_5_x86_64.manylinux1_x86_64.whl
```

Comment (clbarnes) [reply .github/workflows/ci.yaml:81]: Aha! That would be great, and probably solve the secondary issues raised in #9 .

**Comment by clbarnes** [comment]
When I try to build this locally, I get

```
⚠  Warning: No compatible manylinux tag found, using the linux tag instead. You won't be able to upload those wheels to pypi.
```

This could relate to my own build environment and not be an issue in the GHA container - is there any way to test that besides trying for a deploy?

**Comment by messense** [comment]
Try use the `konstin2/maturin` docker image.

**Comment by messense** [comment]
Here is a test on my fork: https://github.com/messense/ncollpyde/runs/2782729128?check_suite_focus=true

**Comment by clbarnes** [comment]
Looks good, thank you for doing that test.

**Pull request merged**